src/routes/ListOfAllArticles.jsx

```diff
@@ -4,26 +4,26 @@ import ArticleSummary from '../components/ArticleSummary'
 import { getAllArticles } from '../../utils/utils'
 import '../css/list_of_articles.css'
 
-export default function ListOfAllArticles() {
+export default function ListOfAllArticles({topic_name}) {
     const [listOfArticles, setListOfArticles] = useState([]);
     const [isLoading, setIsLoading] = useState(true)
 
     useEffect(() => {
-        getAllArticles().then(({articles}) => setListOfArticles(articles)).then(() => setIsLoading(false))
-    }, []);
+        getAllArticles(topic_name).then(({articles}) => setListOfArticles(articles)).then(() => setIsLoading(false))
+    }, [listOfArticles]);
```

Comment:
You are setting listOfArticles, and you are using that state as a dependency as well, which has resulted in an infinite loop. Please solve that issue.